snowfakery/data_generator_runtime.py

```diff
@@ -235,22 +227,44 @@ def check_slots_filled(self):
         if not_filled:
             plural = "s" if len(not_filled) > 1 else ""
             raise DataGenNameError(
-                f"Reference{plural} not fulfilled: {','.join(not_filled)}",
-                None,
-                None,
+                f"Reference{plural} not fulfilled: {','.join(not_filled)}", None, None,
             )
 
     def __getstate__(self):
-        state = self.__dict__.copy()
-        del state["intertable_dependencies"]
-        del state["named_slots"]
-        return state
+        id_manager = self.id_manager
+        last_seen_obj_of_type = {
+            k: v.__getstate__() for k, v in self.last_seen_obj_of_type.items()
+        }
+        nicknamed_objects = {
+            k: v.__getstate__() for k, v in self.nicknamed_objects.items()
+        }
+        nicknames_and_tables = {k: v for k, v in self.nicknames_and_tables.items()}
+        today = self.today
+        intertable_dependencies = [
+            dict(v._asdict()) for v in self.intertable_dependencies
+        ]  # converts ordered-dict to dict for Python 3.6 and 3.7
+        return {
+            "id_manager": id_manager.__getstate__(),
+            "last_seen_obj_of_type": last_seen_obj_of_type,
+            "nicknamed_objects": nicknamed_objects,
+            "nicknames_and_tables": nicknames_and_tables,
+            "today": today,
+            "intertable_dependencies": intertable_dependencies,
+        }
 
-    def __setstate__(self, state):
-        self.last_seen_obj_of_type = {}
-        for slot, value in state.items():
-            setattr(self, slot, value)
-        self.intertable_dependencies = set()
+    def __setstate__(self, dict):
```

Comment:
Call it d instead of dict to avoid shadowing a builtin

Reply:
49fe14d